### ISSUE TITLE
[MVP] implement crypto / identity part of the protocol

### ISSUE THREAD

**Issue by aeddi** [berty/berty#1303]
Depends on #1301

- [ ] Implement the sigchain, identity generation for account, device, etc...
- [ ] Implement the public RDV point system
- [ ] Implement the "add contact" handshake (as described here: https://github.com/berty/protocol/tree/master/docs#handshakes6)
- [ ] Implement the "already a contact" handshake
- [ ] Define if we have enough time to implement device-to-device double ratchet for the MVP or if it's postponed
- [ ] Implement the device-to-device post-handshake communication logic

Required time (estimation): 21d
